server/src/main/java/org/elasticsearch/action/support/master/ShardsAcknowledgedResponse.java

```diff
@@ -42,10 +42,15 @@
                 ObjectParser.ValueType.BOOLEAN);
     }
 
-    private boolean shardsAcknowledged;
-
-
-    protected ShardsAcknowledgedResponse() {
+    private final boolean shardsAcknowledged;
+
+    protected ShardsAcknowledgedResponse(StreamInput in, boolean readShardsAcknowledged) throws IOException {
+        super(in);
+        if (readShardsAcknowledged) {
```

Comment:
are there tests for this?

Reply:
I pushed: 9351c3d3b6f1a546e7aa9d54add798549821ef67